constants/models.ts

```diff
@@ -0,0 +1,205 @@
+    type: 'GPT-4',
+    tokens: '8,192 tokens',
+    compatibility: 'chat',
+    strengths: 'Function calling data, snapshot model'
+  },
+  {
+    id: 'gpt-4-32k',
+    name: 'gpt-4-32k',
+    description:
+      'Same capabilities as the base gpt-4 model but with 4x the context length.',
+    type: 'GPT-4',
+    compatibility: 'chat',
+    tokens: '32,768 tokens',
+    strengths: 'Extended context capabilities'
+  },
+  {
+    id: 'gpt-4-32k-0613',
+    name: 'gpt-4-32k-0613',
+    description:
+      'Snapshot of gpt-4-32 from June 13th 2023. This model will not receive updates, and will be deprecated 3 months after a new version is released.',
+    type: 'GPT-4',
+    compatibility: 'chat',
+    tokens: '32,768 tokens',
+    strengths: 'Extended context capabilities, snapshot model'
+  },
+
+  {
+    id: 'text-curie-001',
+    name: 'text-curie-001',
+    description: 'Very capable, faster and lower cost than Davinci.',
+    type: 'GPT-3',
+    compatibility: 'completions',
+    tokens: '2,049 tokens',
+    strengths: 'High capability, speed, cost-effectiveness'
+  },
+  {
+    id: 'text-babbage-001',
+    name: 'text-babbage-001',
+    description: 'Capable of straightforward tasks, very fast, and lower cost.',
+    type: 'GPT-3',
+    compatibility: 'completions',
+    tokens: '2,049 tokens',
+    strengths: 'Straightforward tasks, speed, cost-effectiveness'
+  },
+  {
+    id: 'text-ada-001',
+    name: 'text-ada-001',
+    description:
+      'Capable of very simple tasks, usually the fastest model in the GPT-3 series, and lowest cost.',
+    type: 'GPT-3',
+    compatibility: 'completions',
+    tokens: '2,049 tokens',
+    strengths: 'Simple tasks, speed, cost-effectiveness'
+  },
+  {
+    id: 'davinci',
+    name: 'davinci',
+    description:
+      'Most capable GPT-3 model. Can do any task the other models can do, often with higher quality.',
+    type: 'GPT-3',
+    compatibility: 'fine-tunes',
+    tokens: '2,049 tokens',
+    strengths: 'High capability, versatile, high quality output'
+  },
+  {
+    id: 'curie',
+    name: 'curie',
+    description: 'Very capable, but faster and lower cost than Davinci.',
+    type: 'GPT-3',
+    compatibility: 'fine-tunes',
+    tokens: '2,049 tokens',
+    strengths: 'High capability, speed, cost-effectiveness'
+  },
+  {
+    id: 'babbage',
+    name: 'babbage',
+    description: 'Capable of straightforward tasks, very fast, and lower cost.',
+    type: 'GPT-3',
+    compatibility: 'fine-tunes',
+    tokens: '2,049 tokens',
+    strengths: 'Straightforward tasks, speed, cost-effectiveness'
+  },
+  {
+    id: 'ada',
+    name: 'ada',
+    description:
+      'Capable of very simple tasks, usually the fastest model in the GPT-3 series, and lowest cost.',
+    type: 'GPT-3',
+    compatibility: 'fine-tunes',
+    tokens: '2,049 tokens',
```

Comment:
This heavily restricts the starter template to OpenAI as it is implemented now. Let's split this into another PR and work through the DX of multiple providers and multiple models more thoroughly.